scenes/game_elements/characters/player/components/player_fighting.gd

```diff
@@ -32,3 +35,11 @@ func _on_body_entered(body: Node2D) -> void:
 
 func _on_air_stream_body_entered(body: Projectile) -> void:
 	body.got_hit(owner)
+
+
+func _notification(what: int) -> void:
+	match what:
+		## When fighting is disabled, reset the modulate that could have been
+		## modifeid in the got_hit animation.
+		NOTIFICATION_DISABLED:
+			player_sprite.modulate.a = 1.0
```

Comment:
This works, but I wonder if we can make use of the RESET animation that Godot creates.
```suggestion
		NOTIFICATION_DISABLED:
			got_hit_animation.play(&"RESET")
```
I'll give it a try.

Reply:
I tried that first and it didn't work! @manuq . I guess it was because the animation player was disabled as well. If you think of a way to playing that before disabling feel free to change it tho!

And btw, in case it helps: for testing this I temporarily changed one of the ink combats to have just 1 ink well with needed_amount = 1, and stretched the got hit animation to 20 seconds so it was a lot easier to reproduce.

Reply:
Mmmm, also, another way of fixing it I'm thinking of right now 🤔 : we could have that animationplayer to have process **_ALWAYS_** instead of inherits, so it will always finish the animation! That should work too, what do you think?

Reply:
Yep, I wasn't able to reproduce the bug by calling RESET. @JuanFdS with your permission, I'm going to edit your commit.

Reply:
> Yep, I wasn't able to reproduce the bug by calling RESET. @JuanFdS with your permission, I'm going to edit your commit.

are you sure? @manuq , I applied that change and was able to reproduce it:

https://github.com/user-attachments/assets/a9deb752-cfe2-400f-ad65-f1ac0669a113

Reply:
> > Yep, I wasn't able to reproduce the bug by calling RESET. @JuanFdS with your permission, I'm going to edit your commit.
> 
> are you sure? @manuq , I applied that change and was able to reproduce it:
> 
>  Grabacion.De.Pantalla.2025-05-09.183657.mp4

Ohh! Let me check...

Reply:
@JuanFdS thanks for catching the error. I made it work by calling `got_hit_animation.advance(0)`. From the documentation of AnimationPlayer, this is needed, otherwise it plays in the next processing, which won't happen for a disabled node.

I really prefer using RESET than doing what the reset animation does, because the animation may change to animate other properties, and then we'll have to change the script.

Yes, I think the other alternative is to set the AnimationPlayer node to always process. This works fine though. I confirmed by changing "got hit" to last 10 seconds and stay transparent.

[Grabación de pantalla desde 2025-05-09 19-07-03.webm](https://github.com/user-attachments/assets/ff5d30dd-dc6a-4d33-b52b-6ef5a52d1d9f)